data/src/main/kotlin/com/vultisig/wallet/data/common/StringExtensions.kt

```diff
@@ -29,6 +30,37 @@ fun String.toByteStringOrHex(): ByteString {
         this.toByteString()
     }
 }
+
+fun String.normalizeMessageFormat(): String {
+    return try {
+        if (this.isHex()) {
+            val hex = this.remove0x()
+            if (hex.length % 2 != 0) {
+                return this
+            }
+            val bytes = hex.chunked(2)
+                .map { it.toInt(16).toByte() }
+                .toByteArray()
+            String(
+                bytes,
+                Charsets.UTF_8
+            ).replace(
+                // Remove leading/trailing control characters
+                "^\\p{C}+|\\p{C}+$".toRegex(),
+                ""
+            )
```

Comment:
@aminsato can you check the following functions and extension functions:

- String.toHexBytes() 
- Numeric.hexStringToByteArray()
- toHexString()

Probably you can reuse those